libshvcore/src/utils/shvfilejournal.cpp

```diff
@@ -28,69 +26,28 @@ using namespace shv::chainpack;
 
 namespace shv::core::utils {
 
-#if defined(__APPLE__)
-#define SHV_STATBUF              struct stat
-#define SHV_STAT                 ::stat
-#else
-#define SHV_STATBUF              struct stat64
-#define SHV_STAT                 ::stat64
-#endif
-
-#if defined(__unix) || defined(__APPLE__)
-#define SHV_MKDIR(dir_name)      ::mkdir(dir_name, 0777)
-#else
-#define SHV_MKDIR(dir_name)      ::mkdir(dir_name)
-#endif
-#define SHV_REMOVE_FILE          ::remove
-
 static bool is_dir(const std::string &dir_name)
 {
-	SHV_STATBUF st;
-	return SHV_STAT(dir_name.data(), &st) == 0 && (st.st_mode & S_IFMT) == S_IFDIR;
+	return std::filesystem::is_directory(dir_name);
 }
 
 static bool mkpath(const std::string &dir_name)
 {
-	// helper function to check if a given path is a directory, since mkdir can
-	// fail if the dir already exists (it may have been created by another
-	// thread or another process)
-	shvDebug() << dir_name << "exists:" << is_dir(dir_name);
-	if(is_dir(dir_name))
-		return true;
-	logIShvJournal() << "Creating journal dir:" << dir_name;
-	if (SHV_MKDIR(dir_name.data()) == 0)
-		return true;
-	if (errno == EEXIST)
-		return is_dir(dir_name);
-	if (errno != ENOENT)
-		return false;
-	// mkdir failed because the parent dir doesn't exist, so try to create it
-	auto const slash = dir_name.find_last_of('/');
-
-	if (slash == std::string::npos)
-		return false;
-	std::string parent_dir_name = dir_name.substr(0, slash);
-	if (!mkpath(parent_dir_name))
-		return false;
-	// try again
-	if (SHV_MKDIR(dir_name.data()) == 0)
-		return true;
-	return errno == EEXIST && is_dir(dir_name);
+	std::error_code code;
+	return std::filesystem::create_directories(dir_name, code);
 }
 
 static int64_t file_size(const std::string &file_name)
 {
-	SHV_STATBUF st;
-	if(SHV_STAT(file_name.data(), &st) == 0)
-		return st.st_size;
-	logWShvJournal() << "Cannot stat file:" << file_name;
-	return -1;
+	std::error_code code;
+	return std::filesystem::file_size(file_name, code);
```

Comment:
we are silently ignoring FS errors, add some logging pls

Reply:
done